digitalocean/resource_digitalocean_database_cluster.go

```diff
@@ -38,7 +39,9 @@ func resourceDigitalOceanDatabaseCluster() *schema.Resource {
 			},
 
 			"version": {
-				Type:     schema.TypeString,
+				Type: schema.TypeString,
+				// In practice, this is required
+				// See transitionVersionToRequired CustomizeDiffFunc
 				Optional: true,
```

Comment:
Did you mean to make this required?

Reply:
Setting `Required: true,` here makes it error out before the `CustomizeDiff` functions are ever run. So in `transitionVersionToRequired()` I handle both cases:

```
if !hasVersion {
	if old != "" {
		return fmt.Errorf(`The argument "version" is now required. Set the %v version to the value saved to state: %v`, engine, old)
	}

	return fmt.Errorf(`The argument "version" is required, but no definition was found.`)
}
```

For a new resource, if `version` is not set this error is returned:

> The argument "version" is required, but no definition was found.

If no version is set but there is a diff (e.g. new: "8" -> old: ""), then you get the error with the hint:

> The argument "version" is now required. Set the $engine version to the value saved to state: $version